File: src/test/java/net/nlt/matrix/multiplication/CommonUtilsTest.java
package net.nlt.matrix.multiplication;

import org.junit.jupiter.api.Test;

import static net.nlt.matrix.multiplication.CommonUtils.nextMultiple;
import static org.assertj.core.api.Assertions.assertThat;

class CommonUtilsTest {

    @Test
    void testAlignSize() {
        assertThat(nextMultiple(0, 4)).isEqualTo(4);

        assertThat(nextMultiple(3, 4)).isEqualTo(4);
        assertThat(nextMultiple(4, 4)).isEqualTo(4);

        assertThat(nextMultiple(5, 4)).isEqualTo(8);
        assertThat(nextMultiple(6, 4)).isEqualTo(8);
        assertThat(nextMultiple(7, 4)).isEqualTo(8);
        assertThat(nextMultiple(8, 4)).isEqualTo(8);
    }
}

File: src/test/kotlin/net/nlt/matrix/multiplication/CommonUtilsTest.kt
package net.nlt.matrix.multiplication

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

internal class CommonUtilsTest {

    @Test
    fun testAlignSize() {
        Assertions.assertThat(CommonUtils.nextMultiple(0, 4)).isEqualTo(4)
        Assertions.assertThat(CommonUtils.nextMultiple(3, 4)).isEqualTo(4)
        Assertions.assertThat(CommonUtils.nextMultiple(4, 4)).isEqualTo(4)
        Assertions.assertThat(CommonUtils.nextMultiple(5, 4)).isEqualTo(8)
        Assertions.assertThat(CommonUtils.nextMultiple(6, 4)).isEqualTo(8)
        Assertions.assertThat(CommonUtils.nextMultiple(7, 4)).isEqualTo(8)
        Assertions.assertThat(CommonUtils.nextMultiple(8, 4)).isEqualTo(8)
    }
}